lib/acts_as_taggable_on_mongoid/taggable/tag_type_definition.rb

```diff
@@ -155,59 +148,40 @@ def add_context_tags_method
       def add_tag_list
         add_list_getter
         add_list_setter
+        add_tag_list_from
+        add_tagger_tag_list
+        add_tagger_tag_lists
         add_all_list_getter
         add_list_exists
         add_list_change
         add_list_changed
         add_changed_from_default?
         add_will_change
         add_get_was
+        add_get_lists_was
+        add_tagger_get_was
         add_reset_list
         add_reset_to_default
       end
 
-      def add_list_getter
-        tag_definition = self
-        tag_list_name  = tag_definition.tag_list_name
-
-        owner.taggable_mixin.module_eval do
-          define_method(tag_list_name) do
-            tag_list_on tag_definition
-          end
-
-          alias_method "#{tag_list_name}_before_type_cast".to_sym, tag_list_name.to_sym
-        end
-      end
-
-      def add_list_setter
-        tag_definition = self
-
-        owner.taggable_mixin.module_eval do
-          define_method("#{tag_definition.tag_list_name}=") do |new_tags|
-            dup_tags        = Array.wrap(new_tags).dup
-            options         = dup_tags.extract_options!.dup
-            options[:parse] = options.fetch(:parse) { true }
-
-            new_list = tag_definition.parse(*dup_tags, options)
-
-            mark_tag_list_changed(new_list)
-            tag_list_set(new_list)
-          end
-        end
-      end
+      private
 
-      def add_all_list_getter
-        tag_definition = self
+      def extract_tag_definition_options(options)
```

Comment:
Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)